### DISCUSSION TITLE
buffer type configuration issue

### DISCUSSION THREAD

**Discussion by hemanthofficial3009** [vectordotdev/vector#12746] [Q&A] [answered]
Hello Team,
We have configured the vector sink instance to buffer on the disk and set the buffer.max_size value and also the batch.max_bytes. For some reason we are facing the following error on the arm platform of vector and not on the x86_64 platform : 
![image](https://user-images.githubusercontent.com/65534235/168624917-f55e2bff-a392-45fa-84a7-f92f4a78cfc7.png)
** error=unknown variant `disk`, expected `memory`**

the version of the vector arm variant we are using is 18.0. Can you help us understand the issue!

**Comment by jszwedko** [comment, marked as answer]
Hi @hemanthofficial3009 !

Can you try a newer version of Vector? I believe we used to not compile in disk buffer support for armv7 but recent Vector builds should have it. Maybe try 0.21.1?

**Reply by bmallya** [reply]
@jszwedko, we seem to have also hit the "libc Vector doesn't run on armv7 #11970"  issue here and are not able to use the latest version.

**Reply by bmallya** [reply]
I tried to do a local compile using cross, but see #12751

**Reply by jszwedko** [reply]
Ah weird, could you try the musl build for armv7 just to see if that works for you? I realize you want to use the libc build, just curious if you see the same config error on newer versions of Vector.

**Reply by bmallya** [reply]
The musl build does not work on the environment that we have unfortunately.

**Reply by bmallya** [reply]
Thanks for the update. We used the latest version of vector(musl) and vector started without any error. I do have a follow up question on buffer, for which I will open a new thread.
